### PR TITLE
gRPC CLI list command

### PR DESCRIPTION
**Pull request by y-zeng**
Add list command for gRPC CLI. 

```
grpc_cli ls <address> [<service>[/<method>]]
    <address>                ; host:port
    <service>                ; Exported service name
    <method>                 ; Method name
    --l                      ; Use a long listing format
```

### CODE REVIEW

File: test/cpp/util/grpc_tool.cc [hunk 1]
```diff
@@ -257,6 +260,105 @@ bool GrpcTool::Help(int argc, const char** argv, const CliCredentials cred,
...
+
+  // If no service is specified, dump the list of services.
+  grpc::string output;
+  if (argc < 2) {
+    // List all services, if --l is passed, then include full description,
+    // otherwise include a summarized list only.
+    if (FLAGS_l) {
+      output = DescribeServiceList(service_list, desc_pool);
+    } else {
+      for (auto const& service : service_list) {
```
Comment (yang-g) [inline test/cpp/util/grpc_tool.cc]: Better not use this: https://github.com/grpc/grpc/blob/master/doc/cpp-style-guide.md

**Comment by yang-g** [comment]
LGTM

Sorry for the delay.

**Comment by y-zeng** [comment]
Thanks a lot for the review!

File: test/cpp/util/service_describer.h [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ */
+
+#include <google/protobuf/descriptor.h>
+#include <grpc++/support/config.h>
+
+namespace grpc {
+namespace testing {
+
+grpc::string DescribeServiceList(std::vector<grpc::string> service_list,
+                                 google::protobuf::DescriptorPool& desc_pool);
```
Comment (yang-g) [inline test/cpp/util/service_describer.h]: Do you want to use grpc:: versions here as well?

Comment (y-zeng) [inline test/cpp/util/service_describer.h]: Fixed.

File: test/cpp/util/grpc_tool.cc (hunk 1 above)
Comment (y-zeng) [inline test/cpp/util/grpc_tool.cc]: Fixed.

**Comment by yang-g** [comment]
LGTM

**Comment by yang-g** [comment]
Maybe try clang-format and push again.

**Comment by y-zeng** [comment]
Test failures are irrelevant. Only `grpc_tool_test` is affected.

**Pull request merged**